ha/core/controllers/pcs/node_controller.py

```diff
@@ -176,6 +167,77 @@ def status(self, nodeids: list = None) -> dict:
         """
         raise HAUnimplemented("This operation is not implemented.")
 
+    def check_cluster_feasibility(self, nodeid: str) -> dict:
+        """
+            Check whether the cluster is going to be offline after node with nodeid is stopped.
+        Args:
+            nodeid (str): Private fqdn define in conf store.
+        Returns:
+            Dictionary : {"status": "", "msg":""}
+        """
+        self._is_node_in_cluster(node_id=nodeid)
+        offline_nodes, node_list = self._get_offline_nodes()
+        if (len(offline_nodes) + 1) >= ((len(node_list)//2) + 1):
+            return {"status": const.STATUSES.FAILED.value, "output": "", "error": "Stopping the node will cause a loss of the quorum"}
+        else:
+            return {"status": const.STATUSES.SUCCEEDED.value, "output": "", "error": ""}
+
+    def _get_offline_nodes(self):
```

Comment:
we have `nodes_status`
in pcs controller can we use this fun to get offline nodes

Reply:
Used node_status() to get offline nodes.